packages/hydrooj/src/service/server.ts

```diff
@@ -417,8 +417,10 @@ export function Connection(
                 h.compression = new Shorty();
                 conn.send('shorty');
             }
+            conn.pause();
             if (h._prepare) await h._prepare(args);
             if (h.prepare) await h.prepare(args);
+            conn.resume();
```

Comment:
suggest to move this after set `conn.onmessage`

Reply:
Although it should be a synced operation from here until `await bus.parallel`, I'll move it down for better readability.